backend/routes/resetCourts.js

```diff
@@ -0,0 +1,97 @@
+const express = require("express");
+const router = express.Router();
+const admin = require("firebase-admin");
+const bcrypt = require("bcrypt");
+
+function bcryptCompare(plaintext, hashed) {
+  return new Promise(function (resolve, reject) {
+    bcrypt.compare(plaintext, hashed, function (err, res) {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(res);
+      }
+    });
+  });
+}
+
+async function resetActivePlayers(location) {
+  const active_players = await location
+    .collection("activePlayers")
+    .listDocuments();
+
+  for (var j = 0; j < active_players.length; j++) {
+    const player_ref = await location
+      .collection("activePlayers")
+      .doc(active_players[j].id);
+    const player = await (await player_ref.get()).data();
+
+    const court_number = player.courtNumber;
+    const new_name = "Empty" + court_number.toString();
+    player.playerWaiting = false;
+    player.firebaseUID = new_name;
+    player.nickname = new_name;
+    // delete the old ref if the new name is not the same as the old id (i.e, clean up old player)
+    if (new_name != active_players[j].id) {
+      await location.collection("activePlayers").doc(new_name).set(player);
+      await player_ref.delete();
+    } else {
+      await player_ref.update({
+        playerWaiting: player.playerWaiting,
+        firebaseUID: player.firebaseUID,
+        nickname: player.nickname,
+      });
+    }
+  }
+}
+
+async function deleteQueuePlayers(location) {
+  const queue_players = await location
+    .collection("queuePlayers")
+    .listDocuments();
+
+  for (var j = 0; j < queue_players.length; j++) {
+    const player_ref = await location
+      .collection("queuePlayers")
+      .doc(queue_players[j].id);
+    await player_ref.delete();

```

Comment:
Can we consider performing a batch delete to call Firestore once? And ensure atomicity so all deletes happen or none do:

const batch = firestore().batch()

queueSnapshot.forEach((doc) => {
    batch.delete(doc.ref);
  });

await batch.commit();